apps/api/src/app/order/order.service.ts

```diff
@@ -123,10 +123,29 @@ export class OrderService {
       data.SymbolProfile.connectOrCreate.create.assetSubClass = assetSubClass;
       data.SymbolProfile.connectOrCreate.create.dataSource = dataSource;
       data.SymbolProfile.connectOrCreate.create.name = name;
-      data.SymbolProfile.connectOrCreate.create.symbol = id;
+
+      let symbolProfileSymbol = id;
+
+      if (data.type === 'INTEREST') {
+        const symbolProfile = await this.prismaService.symbolProfile.findFirst({
+          orderBy: {
+            createdAt: 'desc'
+          },
+          where: {
```

Comment:
There will be collisions in the cloud setup. A `userId` (including migration path) is needed to avoid them.

Reply:
Add `currency` as below?

Reply:
> There will be collisions in the cloud setup. A `userId` (including migration path) is needed to avoid them.

You mean collisions like users A and B having the same INTEREST named "Bank Interest"? Shouldn't it be the fine, in the same way that it's fine for users A and B to have different activities pertaining the same AAPL symbol, for instance?

Reply:
Maybe it would work for `INTEREST`, but I see an issue for `ITEM`. If you and me both own a _Beach House_, we could not share the same asset profile because of different historical market prices.

Reply:
Ah, good point! But in that same train of thought, I could see a user having multiple `ITEM` items with the same name to represent different physical items, right? Should we keep `ITEM`s as is so as to not break that expected behavior for historical market prices?

Reply:
Yes, we could keep `ITEM` (and `LIABILITY`) as is and see them as unique.
As a result there will be one Symbol Profile per `INTEREST` / `FEE`?

Reply:
Sounds good to me, I'll update accordingly.
Just to confirm though - do you still think we should have `INTEREST`/`FEE` unique per `userId` or it's ok for them to be globally unique?

Reply:
I would still do it unique per `userId`, otherwise if I rename it would also be renamed for you.

Reply:
So, in the current implementation that would not happen. See `updateOrder` below. When you rename a `INTEREST` order, it will either connect it to an existing `SymbolProfile` with the new name or create a new `SymbolProfile` if one doesn't exist with that name.